src/main/kotlin/blackjack/domain/state/GameResult.kt

```diff
@@ -1,32 +1,15 @@
 package blackjack.domain.state
 
-import blackjack.domain.player.Player
-
-class GameResult(private val dealer: Player, private val users: List<Player>) {
-    private val dealerResult: String
-
-    init {
-        val winCount =
-            users.filter {
-                dealer.comparePoints(it) == ResultState.WIN
-            }.size
-        val loseCount =
-            users.filter {
-                dealer.comparePoints(it) == ResultState.LOSE
-            }.size
-        val drawCount = users.size - winCount + loseCount
-        dealerResult = "${winCount}승 ${drawCount}무 ${loseCount}패"
-    }
-
-    fun getDealerResult(): String {
-        return dealerResult
-    }
-
-    fun getGameUserResult(): List<String> {
-        val results =
-            users.map {
-                "${it.name}: ${dealer.comparePoints(it).not().displayMessage}"
-            }
-        return results
-    }
+import blackjack.domain.player.Dealer
+import blackjack.domain.player.GameUser
+
+class GameResult(dealer: Dealer, gameUsers: List<GameUser>) {
+    var dealerBalance = 0
+        private set
+    val usersResult: Map<GameUser, ResultState> =
+        gameUsers.associateWith {
+            val userResultState = dealer.compareGetResultOpponent(it)
+            dealerBalance -= it.bettingRevenue(userResultState)
+            userResultState
+        }
```

Comment:
```suggestion
    val usersResult: Map<GameUser, ResultState> = gameUsers.associateWith {
        dealer.compareGetResultOpponent(it)
    }

    val dealerBalance: Int = usersResult.map { (user, result) ->
        user.bettingRevenue(result) * -1
    }.sum()
```

물론 지금 코드는 가변 변수로도 충분히 깔끔하게 구현이 가능하지만 계산하는데 큰 코스트가 드는게 아니라면 불변 객체를 만드는데 초점을 둬서 구현해보시는 것도 좋을 것 같습니다!
클래스와 코드의 크기가 커질수록 가변 객체는 예측하기 어려워집니다. 교육 과정 중에는 평소와 다른 스타일을 시도해보시는 것도 좋을 것 같아요 😄

Reply:
훨씬 깔끔하고 보기 좋은 코드네요
아직 멀었군요 :)